compiler/transforms/stream_snaxify.py

```diff
@@ -17,6 +18,40 @@
 from compiler.dialects.snax import StreamerConfigurationAttr
 
 
+@dataclass
+class HoistAcceleratorAttribute(RewritePattern):
+    """
+    A pattern to hoist the library call from within a memref
+    streaming region as an attribute of the op.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: memref_stream.StreamingRegionOp, _):
+        # attribute already assigned to op
+        if getattr(op.attributes, "accelerator", None):
+            return
+
+        accelerator = None
+
+        for inner_op in op.body.walk():
+            # look for memref_stream.generic with library call
+            if not isinstance(inner_op, memref_stream.GenericOp):
+                continue
+
+            if not inner_op.library_call:
+                continue
+
+            if accelerator and inner_op.library_call != accelerator:
+                raise RuntimeError(
+                    "multiple different accelerator dispatches found in streaming region op"
```

Comment:
```suggestion
                    "multiple different accelerator dispatches found in single memref_stream.streaming_region op"
```
Can you add a test to check if this error is thrown?

Reply:
This seems to be a bit uncomfortable. xdsl testing system does not really allow for checking runtime errors in filechecks. Specific exceptions are made for parsing errors and verify errors, but this is neither. In all other passes, when runtime errors are thrown, they aren't really checked 😬 (for example: insert-accfg-op)

Reply:
oh, allright then :)

Reply:
Yes!